src/bun.js/bindings/KeyObject.cpp

```diff
@@ -788,7 +788,7 @@ static JSC::EncodedJSValue KeyObject__createOKPFromPrivate(JSC::JSGlobalObject*
     Vector<unsigned char> public_key(ED25519_PUBLIC_KEY_LEN);
 
     if (namedCurve == CryptoKeyOKP::NamedCurve::Ed25519) {
-        memcpy(public_key.data(), keyData.data() + ED25519_PRIVATE_KEY_LEN, ED25519_PUBLIC_KEY_LEN);
+        memcpy(public_key.data(), keyData.data(), ED25519_PUBLIC_KEY_LEN);
```

Comment:
We should add a `publicKeyData()` method on `WebCore::CryptoKeyOKP::KeyMaterial` that returns a span of the right byte size for the value and calls this

```
static constexpr size_t internalKeySizeInBytesFromNamedCurve(CryptoKeyOKP::NamedCurve curve, CryptoKeyType type)
{
    switch (curve) {
    case CryptoKeyOKP::NamedCurve::X25519:
        return 32;
    case CryptoKeyOKP::NamedCurve::Ed25519:
        return type == CryptoKeyType::Private ? 64 : 32;
    default:
        return -1;
    }
}

```